control/src/lib.rs

```diff
@@ -477,8 +475,17 @@ pub mod pallet {
 			who: T::AccountId
 		) -> DispatchResultWithPostInfo {
 			let org = Orgs::<T>::get(&org_id).ok_or(Error::<T>::OrganizationUnknown)?;
-			Self::ensure_root_or_prime(origin, org.prime.clone(), org.org_type)?;
-
+			match org.access_model {
```

Comment:
Probably it would be good idea to move this checks to separate method, like `ensure_has_modify_access(org, origin, who)`, which will perform all the checks in one place? Then you can just call that method here and in `remove_member` instead of duplicating this checks in both places?